### ISSUE TITLE
ESP32: Building fails when the "build" directory does not exist

### ISSUE THREAD

**Issue by robert-hh** [micropython/micropython#4433]
Building the image with the new set-up for the esp-idf fails, when the build directory does not exist. I get the error message below.
```
Use make V=1 or set BUILD_VERBOSE in your environment to increase build verbosity.
GEN build/sdkconfig.h
Traceback (most recent call last):
  File "/home/robert/Downloads/esp-idf/tools/kconfig_new/confgen.py", line 299, in <module>
    main()
  File "/home/robert/Downloads/esp-idf/tools/kconfig_new/confgen.py", line 104, in main
    update_if_changed(temp_file, filename)
  File "/home/robert/Downloads/esp-idf/tools/kconfig_new/confgen.py", line 277, in update_if_changed
    with open(destination, "w") as f:
IOError: [Errno 2] No such file or directory: 'build/sdkconfig.h'
Makefile:236: die Regel für Ziel „build/sdkconfig.h“ scheiterte
make: *** [build/sdkconfig.h] Fehler 1
```

**Comment by dpgeorge** [comment]
Oops, I think I missed a mkdir dependency.  Try `make -j2` or `mkdir build; make`.

**Comment by dpgeorge** [comment]
Thanks for reporting this, should be fixed by 5d88272342d134bec030af313188ff929aa56dd3
